From 0182c2941509a11a0cd058364b9cb54d1d152954 Mon Sep 17 00:00:00 2001
From: Fred Chen <Fred.Chen@quantatw.com>
Date: Fri, 19 Dec 2025 08:56:05 +0800
Subject: [PATCH 1/3] configurations: Revise the OCP NIC sensor name

Apply a conversion formula to map I2C bus numbers to NIC IDs. The
formula needs to work with two different I2C bus groups.

I2C Bus Group1: [11, 92, 93, 95, 96]
I2C Bus Group2: [11, 100, 101, 103, 104]
Mapping NIC number: [0, 1, 2, 3, 4]

Tested on Santabarbara
```
root@bmc:~# mfg-tool sensor-display 2>/dev/null| table-sensor-display | grep -i nic
NIC0_TEMP_C                   	ok         	37        	DegreesC	N/A	5          	N/A        	N/A	105        	N/A
NIC2_TEMP_C                   	ok         	75        	DegreesC	N/A	5          	N/A        	N/A	105        	N/A
NIC3_TEMP_C                   	ok         	59        	DegreesC	N/A	5          	N/A        	N/A	105        	N/A
```

Change-Id: I6c95ecebe8011d2c01c72886931e3aac147c019c
---
 configurations/broadcomm/400g_1p_ocp3.json | 8 ++++----
 configurations/ocp/cx7_ocp.json            | 8 ++++----
 configurations/ocp/cx8_ocp.json            | 8 ++++----
 3 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/configurations/broadcomm/400g_1p_ocp3.json b/configurations/broadcomm/400g_1p_ocp3.json
index 127f0bb..53074c6 100644
--- a/configurations/broadcomm/400g_1p_ocp3.json
+++ b/configurations/broadcomm/400g_1p_ocp3.json
@@ -3,7 +3,7 @@
         {
             "Address": "$address",
             "Bus": "$bus",
-            "Name": "BRCM OCP NIC FRU $bus",
+            "Name": "BRCM OCP NIC FRU $bus % 109 % 11",
             "Type": "EEPROM"
         },
         {
@@ -12,8 +12,8 @@
             "Labels": [
                 "temp2"
             ],
-            "Name": "NIC Temp $bus",
-            "Name1": "NIC Temp $bus",
+            "Name": "NIC$bus % 109 % 11 TEMP_C",
+            "Name1": "NIC$bus % 109 % 11 TEMP_C",
             "Thresholds": [
                 {
                     "Direction": "less than",
@@ -33,7 +33,7 @@
             "Type": "TMP421"
         }
     ],
-    "Name": "BRCM OCP NIC FRU $bus",
+    "Name": "BRCM OCP NIC FRU $bus % 109 % 11",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'BRCM 400G 1P.*57608 OCP3', 'BOARD_MANUFACTURER': 'Broadcom'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
diff --git a/configurations/ocp/cx7_ocp.json b/configurations/ocp/cx7_ocp.json
index 3f11519..f9c1965 100644
--- a/configurations/ocp/cx7_ocp.json
+++ b/configurations/ocp/cx7_ocp.json
@@ -3,7 +3,7 @@
         {
             "Address": "$address",
             "Bus": "$bus",
-            "Name": "OCP CX7 NIC FRU $bus",
+            "Name": "OCP CX7 NIC FRU $bus % 109 % 11",
             "Type": "EEPROM"
         },
         {
@@ -12,8 +12,8 @@
             "Labels": [
                 "temp2"
             ],
-            "Name": "NIC Temp $bus",
-            "Name1": "NIC Temp $bus",
+            "Name": "NIC$bus % 109 % 11 TEMP_C",
+            "Name1": "NIC$bus % 109 % 11 TEMP_C",
             "Thresholds": [
                 {
                     "Direction": "less than",
@@ -33,7 +33,7 @@
             "Type": "TMP421"
         }
     ],
-    "Name": "OCP CX7 NIC $bus",
+    "Name": "OCP CX7 NIC $bus % 109 % 11",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Nvidia ConnectX-7 OCP3.0'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
diff --git a/configurations/ocp/cx8_ocp.json b/configurations/ocp/cx8_ocp.json
index baa4359..f858767 100644
--- a/configurations/ocp/cx8_ocp.json
+++ b/configurations/ocp/cx8_ocp.json
@@ -3,7 +3,7 @@
         {
             "Address": "$address",
             "Bus": "$bus",
-            "Name": "OCP CX8 NIC FRU $bus",
+            "Name": "OCP CX8 NIC FRU $bus % 109 % 11",
             "Type": "EEPROM"
         },
         {
@@ -12,8 +12,8 @@
             "Labels": [
                 "temp2"
             ],
-            "Name": "NIC Temp $bus",
-            "Name1": "NIC Temp $bus",
+            "Name": "NIC$bus % 109 % 11 TEMP_C",
+            "Name1": "NIC$bus % 109 % 11 TEMP_C",
             "Thresholds": [
                 {
                     "Direction": "less than",
@@ -33,7 +33,7 @@
             "Type": "TMP421"
         }
     ],
-    "Name": "OCP CX8 NIC $bus",
+    "Name": "OCP CX8 NIC $bus % 109 % 11",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Nvidia ConnectX-8 OCP3.0'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
-- 
2.47.3

